samples/matter/common/src/bridge/bridge_storage_manager.cpp

```diff
@@ -123,31 +200,85 @@ bool BridgeStorageManager::RemoveBridgedDeviceType(uint8_t bridgedDeviceIndex)
 	return Nrf::GetPersistentStorage().NonSecureRemove(&id);
 }
 
-bool BridgeStorageManager::StoreBridgedDevice(const MatterBridgedDevice *device, uint8_t index)
+bool BridgeStorageManager::LoadBridgedDevice(BridgedDevice &device, uint8_t index)
 {
-	if (!device) {
+	Nrf::PersistentStorageNode id = CreateIndexNode(index, &mBridgedDevice);
+	size_t readSize = 0;
+	uint8_t buffer[sizeof(BridgedDevice) + kMaxUserDataSize];
+	uint16_t counter = 0;
+
+	if (!Nrf::PersistentStorage::Instance().Load(&id, buffer, sizeof(BridgedDevice) + kMaxUserDataSize, readSize)) {
 		return false;
 	}
 
-	if (!StoreBridgedDeviceEndpointId(device->GetEndpointId(), index)) {
-		return false;
+	/* Deserialize data and copy it from buffer into structure's fields. */
+	memcpy(&device.mEndpointId, buffer, sizeof(device.mEndpointId));
+	counter += sizeof(device.mEndpointId);
+	memcpy(&device.mDeviceType, buffer + counter, sizeof(device.mDeviceType));
+	counter += sizeof(device.mDeviceType);
+	memcpy(&device.mNodeLabelLength, buffer + counter, sizeof(device.mNodeLabelLength));
+	counter += sizeof(device.mNodeLabelLength);
+	memcpy(device.mNodeLabel, buffer + counter, device.mNodeLabelLength);
+	counter += device.mNodeLabelLength;
+
+	/* Check if user prepared a buffer for reading user data. It can be nullptr if not needed. */
+	if (!device.mUserData) {
+		device.mUserDataSize = 0;
+		return true;
 	}
 
-	if (!StoreBridgedDeviceNodeLabel(device->GetNodeLabel(), strlen(device->GetNodeLabel()), index)) {
+	/* Validate if user buffer size is big enough to fit the stored user data size. */
+	if (device.mUserDataSize < readSize - counter - sizeof(device.mUserDataSize)) {
```

Comment:
```suggestion
	if (counter + sizeof(device.mUserDataSize) + device.mUserDataSize < readSize) {
```
to avoid integer underflow and always true condition when dealing with unsigned types.

So this assumes a user must know an exact size of the user data. Probably ok for now but for some user data this may be too restrictive.

Reply:
> So this assumes a user must know an exact size of the user data. 

This wasn't quite my intention. The user should know the max acceptable user data size and then the field is updated with data size read from the storage. I understand it could not be obvious, so I updated the method description.